Dockerfile.nanoserver

```diff
@@ -1,17 +1,29 @@
 ARG TAG=ltsc2022
 FROM mcr.microsoft.com/dotnet/sdk:8.0-nanoserver-$TAG AS build
+WORKDIR /src
 
-WORKDIR /source
+# Copy csproj and restore as distinct layers
+COPY libs/client/*.csproj libs/client/
+COPY libs/cluster/*.csproj libs/cluster/
+COPY libs/common/*.csproj libs/common/
+COPY libs/host/*.csproj libs/host/
+COPY libs/server/*.csproj libs/server/
+COPY libs/storage/Tsavorite/cs/src/core/*.csproj libs/storage/Tsavorite/cs/src/core/
+COPY libs/storage/Tsavorite/cs/src/devices/AzureStorageDevice/*.csproj libs/storage/Tsavorite/cs/src/devices/AzureStorageDevice/
+COPY main/GarnetServer/*.csproj main/GarnetServer/
+COPY metrics/HdrHistogram/*.csproj metrics/HdrHistogram/
 
-# Copy files
-COPY . .
-WORKDIR /source/main/GarnetServer
+RUN dotnet restore main/GarnetServer/GarnetServer.csproj
 
-RUN dotnet restore
-RUN dotnet build -c Release
+# Copy everthing else and publish app
+COPY Garnet.snk Garnet.snk
+COPY libs/ libs/
+COPY main/ main/
+COPY metrics/ metrics/
+COPY test/testcerts test/testcerts
 
-# Copy and publish app and libraries
-RUN dotnet publish -c Release -o /app -r win-x64 --self-contained false -f net8.0
+WORKDIR /src/main/GarnetServer
+RUN dotnet publish -c Release -o /app -r win-x64 --no-restore --self-contained false -f net8.0
```

Comment:
The build is still failing for Dockerfile.nanoserver but in a different place now. 
The build command is: docker build -t nanosvr -f dockerfile.nanoserver .
MSBuild version 17.9.6+a4ecab324 for .NET
C:\Program Files\dotnet\sdk\8.0.203\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1047: Assets file 'C:\src\main\GarnetServer\obj\project.assets.json' doesn't have a target for 'net8.0/win-x64'. Ensure that restore has run and that you have included 'net8.0' in the TargetFrameworks for your project. You may also need to include 'win-x64' in your project's RuntimeIdentifiers. [C:\src\main\GarnetServer\garnetserver.csproj::TargetFramework=net8.0]
The command 'cmd /S /C dotnet publish -c Release -o /app -r win-x64 --no-restore --self-contained false -f net8.0' returned a non-zero code: 1

Reply:
Looks like there's a runtime identifier problem between restore and publish operations. I've pushed a fix so that both operations use the current system runtime.